plugins/hydra_ax_sweeper/hydra_plugins/hydra_ax_sweeper/_core.py

```diff
@@ -76,14 +84,20 @@ def get_one_batch_of_trials(
 
     batch_of_trials = []
     for _ in range(max_parallelism_setting):
-        parameters, trial_index = ax_client.get_next_trial()
-        batch_of_trials.append(
-            Trial(
-                overrides=map_params_to_arg_list(params=parameters),
-                trial_index=trial_index,
+        try:
```

Comment:
Since the exception is terminating the search, why not let it fly to the calling function and handle it there?

Reply:
we would loose all the batches that we generated in this function (before the exception was raised).  For example, say the search space exhausted after 5 trials and num_max_trials_to_do = 10. So directly raising the exception means we do not evaluate the first 5 trials.

Reply:
I see, I missed the sweep in between because it was not showing in the diff.